smart_contracts/artifacts/TestingApp/client-ts.generated.ts

```diff
@@ -535,133 +549,170 @@ export class TestingAppClient {
    * @returns The deployment result
    */
   public deploy(params: TestingAppDeployArgs & AppClientDeployCoreParams = {}) {
-    const { boxes: create_boxes, lease: create_lease, onCompleteAction: createOnCompleteAction, ...createArgs } = params.createArgs ?? {}
-    const { boxes: update_boxes, lease: update_lease, ...updateArgs } = params.updateArgs ?? {}
-    const { boxes: delete_boxes, lease: delete_lease, ...deleteArgs } = params.deleteArgs ?? {}
-    return this.appClient.deploy({ 
-      ...params,
-      createArgs: params.createArgs ? this.mapMethodArgs(createArgs, { boxes: create_boxes, lease: create_lease }) : undefined,
-      createOnCompleteAction,
-      updateArgs: params.updateArgs ? this.mapMethodArgs(updateArgs, { boxes: update_boxes, lease: update_lease }) : undefined,
-      deleteArgs: params.deleteArgs ? this.mapMethodArgs(deleteArgs, { boxes: delete_boxes, lease: delete_lease }) : undefined,
-    })
-  }
+  return this.appClient.deploy({ 
+    ...params,
+    createArgs: Array.isArray(params.createArgs) ? mapBySignature(...params.createArgs as [any, any, any]): params.createArgs,
+    deleteArgs: Array.isArray(params.deleteArgs) ? mapBySignature(...params.deleteArgs as [any, any, any]): params.deleteArgs,
+    updateArgs: Array.isArray(params.updateArgs) ? mapBySignature(...params.updateArgs as [any, any, any]): params.updateArgs,
+  })
+}
 
-  /**
-   * Creates a new instance of the TestingApp smart contract.
-   * @param args The arguments for the contract call
-   * @param params Any additional parameters for the call
-   * @returns The creation result
-   */
-  public create<TMethod extends string>(args: { method?: TMethod } & TestingAppCreateArgs = {}, params?: AppClientCallCoreParams & AppClientCompilationParams & CoreAppCallArgs) {
-    const onCompleteAction = args.onCompleteAction
-    return this.mapReturnValue<TMethod>(this.appClient.create({ ...this.mapMethodArgs(args), ...params, ...{ onCompleteAction } }))
+/**
+ * Creates a new instance of the TestingApp smart contract using a bare call.
+ * @param args The arguments for the bare call
+ * @returns The create result
+ */
+public create(args: BareCallArgs & AppClientCallCoreParams & AppClientCompilationParams & CoreAppCallArgs & (OnCompleteNoOp | OnCompleteOptIn)): Promise<AppCallTransactionResultOfType<undefined>>;
+/**
+ * Creates a new instance of the TestingApp smart contract using the create_abi(string)string ABI method.
+ * @param method The ABI method to use
+ * @param args The arguments for the contract call
+ * @param params Any additional parameters for the call
+ * @returns The create result
+ */
+public create(method: 'create_abi(string)string', args: MethodArgs<'create_abi(string)string'>, params?: AppClientCallCoreParams & AppClientCompilationParams  & (OnCompleteNoOp)): Promise<AppCallTransactionResultOfType<MethodReturn<'create_abi(string)string'>>>;
```

Comment:
@tristanmenzel  It would be nice if there isn't a name collision that you don't need to specify the whole method signature?

Reply:
yeah, we can do this - my thoughts are that it will mean a breaking change if someone _adds_ a collision. It could be that we are ok with this.